src/main/java/org/broadinstitute/hellbender/engine/spark/JoinReadsWithVariants.java

```diff
@@ -66,7 +66,15 @@
 
         JavaPairRDD<GATKRead, Variant> allPairs = pairReadsWithVariants(readsWShards, variantsWShards);
 
-        return allPairs.distinct().groupByKey();
+        return allPairs.aggregateByKey(new HashSet<>(), (vs, v) -> {
+            if (v != null) { // pairReadsWithVariants can produce null variant
+                ((HashSet<Variant>) vs).add(v);
```

Comment:
I usually use the Monte Carlo method for fixing issues with generics, and it doesn't seem to be working.  This gives me an error:

```
return allPairs.aggregateByKey(new HashSet<Variant>(), (HashSet<Variant> vs, Variant v) -> {
        if (v != null) { // pairReadsWithVariants can produce null variant
            vs.add(v);
        }
        return vs;
    }, (HashSet<Variant> vs1, HashSet<Variant> vs2) -> {
        vs1.addAll(vs2);
        return vs1;
    })
```

What is the correct change?